src/shared/components/data-table/DataTable.tsx

```diff
@@ -33,12 +33,12 @@ import {
 
 import styles from './DataTable.scss';
 
-type TableContextType = DataTableState & {
+export type TableContextType = DataTableState & {
   dispatch: React.Dispatch<AllTableActions>;
   columns: DataTableColumns;
-  theme: TableTheme;
-  selectableColumnIndex: number;
-  expandedContent: { [rowId: string]: ReactNode };
+  theme?: TableTheme;
+  selectableColumnIndex?: number;
+  expandedContent?: { [rowId: string]: ReactNode };
```

Comment:
Ah, this reminds me. The `ReactNode` type is an output type of a React component; it suggests that React had to prerender the expanded content for _all_ rows before the user even expands a row. I think expanded content should be rendered lazily and on demand rather than all at once beforehand.

Reply:
Currently, `expandedContent` gets updated only on demand.